src/Tribe/Commerce/PayPal/Main.php

```diff
@@ -546,42 +546,49 @@ public function attendance_totals() {
 	 *
 	 * @since 4.7
 	 *
-	 * @param array  $data
-	 * @param string $order_id
-	 * @param int    $event_id
+	 * @param array $attendee_data Information that we are trying to save.
+	 * @param int   $attendee_id   The attendee ID.
+	 * @param int   $post_id       The event/post ID.
 	 */
-	public function update_attendee_data( $data, $order_id, $event_id ) {
+	public function update_attendee_data( $attendee_data, $attendee_id, $post_id ) {
 		$user_id = get_current_user_id();
 
-		$ticket_orders    = $this->tickets_view->get_post_ticket_attendees( $event_id, $user_id );
-		$ticket_order_ids = wp_list_pluck( $ticket_orders, 'order_id' );
+		$ticket_attendees    = $this->tickets_view->get_post_ticket_attendees( $post_id, $user_id );
+		$ticket_attendee_ids = wp_list_pluck( $ticket_attendees, 'attendee_id' );
 
-		// This makes sure we don't save attendees for orders that are not from this current user and event
-		if ( ! in_array( $order_id, $ticket_order_ids ) ) {
+		// This makes sure we don't save attendees for attendees that are not from this current user and event.
+		if ( ! in_array( $attendee_id, $ticket_attendee_ids, true ) ) {
 			return;
 		}
 
-		// Get the Attendee Data, it's important for testing
-		foreach ( $ticket_orders as $test_attendee ) {
-			if ( $order_id !== $test_attendee['order_id'] ) {
-				continue;
-			}
+		$attendee_data_to_save = [];
+
+		// Only update full name if set.
+		if ( ! empty( $attendee_data['full_name'] ) ) {
+			$attendee_data_to_save['full_name'] = sanitize_text_field( $attendee_data['full_name'] );
 		}
 
-		$attendee_email     = empty( $data['email'] ) ? null : sanitize_email( $data['email'] );
-		$attendee_email     = is_email( $attendee_email ) ? $attendee_email : null;
-		$attendee_full_name = empty( $data['full_name'] ) ? null : sanitize_text_field( $data['full_name'] );
-		$attendee_optout    = empty( $data['optout'] ) ? 0 : max( 0, (int) $data['optout'] );
+		// Only update email if set.
+		if ( ! empty( $attendee_data['email'] ) ) {
+			$attendee_data['email'] = sanitize_email( $attendee_data['email'] );
 
-		update_post_meta( $order_id, $this->attendee_optout_key, (int) $attendee_optout );
+			// Only update email if valid
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: d1cc280e1bbe15f6d1aabf9b4cf050c7</sub>